File: project_java/src/main/java/hu/bme/aut/onlab/dxhjoh/couchsurfing_java_backend/controller/UserController.java
package hu.bme.aut.onlab.dxhjoh.couchsurfing_java_backend.controller;

import hu.bme.aut.onlab.dxhjoh.couchsurfing_java_backend.request.PasswordRequest;
import hu.bme.aut.onlab.dxhjoh.couchsurfing_java_backend.request.UserRequest;
import hu.bme.aut.onlab.dxhjoh.couchsurfing_java_backend.response.UserResponse;
import hu.bme.aut.onlab.dxhjoh.couchsurfing_java_backend.service.declaration.UserService;
import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

@Slf4j
@RestController
@RequestMapping("/api/user")
@RequiredArgsConstructor
public class UserController {

    private final UserService userService;

    // saját felhasználó megtekintése
    @GetMapping
    //@Operation(summary = "Saját felhasználó megtekintése")
    public ResponseEntity<UserResponse> getMyUser() {
        log.trace("UserController : getMyUser");
        return ResponseEntity.ok(userService.getMe());
    }

    // saját felhasználó módosítása
    @PutMapping
    //@Operation(summary = "Saját felhasználó módosítása")
    public ResponseEntity<UserResponse> updateMyUser(@RequestBody UserRequest userReq) {
        log.trace("UserController : updateMyUser, userDto=[{}]", userReq);
        return ResponseEntity.ok(userService.updateMe(userReq));
    }

    // saját felhasználó jelszavának módosítása
    @PutMapping("/password")
    //@Operation(summary = "Saját felhasználó jelszavának módosítása")
    public ResponseEntity<Void> updateMyPassword(@RequestBody PasswordRequest passwordReq) {
        log.trace("UserController : updateMyPassword, passwordDto=[{}]", passwordReq);
        userService.password(passwordReq);
        return new ResponseEntity(HttpStatus.NO_CONTENT);
    }

    // saját felhasználó törlése
    @DeleteMapping
    //@Operation(summary = "Saját felhasználó törlése")
    public ResponseEntity<Void>  deleteMyUser(){
        userService.deleteMe();
        return new ResponseEntity(HttpStatus.NO_CONTENT);
    }

    // felhasználó adatainak lekérése a felhasználó id-ja alapján
    @GetMapping("/{id}")
    //@Operation(summary = "Felhasználó adatainak lekérése a felhasználó id-ja alapján")
    public ResponseEntity<UserResponse> getUser(@PathVariable int id) {
        log.trace("UserController : getUser, id=[{}]", id);
        return ResponseEntity.ok(userService.get(id));
    }
}

File: project/src/main/kotlin/hu/bme/aut/onlab/couchsurfing_backend/controller/UserController.kt
package hu.bme.aut.onlab.couchsurfing_backend.controller

import hu.bme.aut.onlab.couchsurfing_backend.dto.PasswordDto
import hu.bme.aut.onlab.couchsurfing_backend.dto.UserDto
import lombok.AllArgsConstructor
import lombok.NoArgsConstructor
import lombok.RequiredArgsConstructor
import lombok.extern.slf4j.Slf4j
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*


@Slf4j
@RestController
@RequestMapping("/api/user")
@NoArgsConstructor
@RequiredArgsConstructor
@AllArgsConstructor
class UserController(private val userService: UserService) {

    // saját felhasználó megtekintése
    @GetMapping
    fun getMyUser(): ResponseEntity<UserDto> {
        return ResponseEntity.ok(userService.getMe())
    }

    // saját felhasználó módosítása
    @PutMapping
    fun updateMyUser(@RequestBody userDto: UserDto): ResponseEntity<UserDto> {
        return ResponseEntity.ok(userService.updateMe(userDto))
    }

    // saját felhasználó jelszavának módosítása
    @PutMapping("/password")
    fun updateMyPassword(@RequestBody passwordDto: PasswordDto): ResponseEntity<Void> {
        userService.password(passwordDto)
        return ResponseEntity(HttpStatus.NO_CONTENT)
    }

    // saját felhasználó törlése
    @DeleteMapping
    fun deleteMyUser(): ResponseEntity<Void> {
        userService.deleteMe()
        return ResponseEntity(HttpStatus.NO_CONTENT)
    }

    // felhasználó adatainak lekérése a felhasználó id-ja alapján
    @GetMapping("/{id}")
    fun getUser(@PathVariable id: Int): ResponseEntity<UserDto> {
        return ResponseEntity.ok(userService.getUser(id))
    }
}